codegen/src/api/mod.rs

```diff
@@ -371,6 +375,43 @@ impl RuntimeGenerator {
             })
         });
 
+        let outer_error_variants = self.metadata.pallets.iter().filter_map(|p| {
+            let variant_name = format_ident!("{}", p.name);
+            let mod_name = format_ident!("{}", p.name.to_string().to_snake_case());
+            let index = proc_macro2::Literal::u8_unsuffixed(p.index);
+
+            p.error.as_ref().map(|_| {
+                quote! {
+                    #[codec(index = #index)]
+                    #variant_name(#mod_name::Error),
+                }
+            })
+        });
+
+        let outer_error = quote! {
+            #default_derives
+            pub enum Error {
+                #( #outer_error_variants )*
+            }
+        };
```

Comment:
Leaving here my thoughts, not necessarily to be done for this PR:

This part of the code seems highly similar to the generation of the outer Event enum.
There might be a few things we could do here to reduce code duplicate, improve reliability and maintenance:
- add a function `generate_outer_enum`
- take in the name of the enum (either Event / Error or other names we might add in the future)
```rust
#[derive(ToTokens)]
enum OuterEnumName {
   Error,
   Event,
}